src/IQueryableObjectSource/DatabaseProvider.cs

```diff
@@ -83,4 +83,24 @@ protected override string ExtractPlanInternal(DbCommand command)
     }
 
     internal override string GetPlanDirectory(string baseDirectory) => Path.Combine(baseDirectory, "Postgres");
+}
+
+class OracleDatabaseProvider(DbCommand command) : DatabaseProvider(command)
+{
+    protected override string ExtractPlanInternal(DbCommand command)
+    {
+        command.CommandText = "EXPLAIN PLAN FOR " + command.CommandText;
+        command.ExecuteNonQuery();
+
+        // Querying the execution plan using DBMS_XPLAN
+        command.CommandText = "SELECT * FROM TABLE(DBMS_XPLAN.DISPLAY())";
```

Comment:
Does this give an actual query plan or an estimated one?

Reply:
Actual Plan

Reply:
This is an estimated plan, should we give option for actual plan or show only actual plan

Reply:
Actual is better as that's how it behaves for other databases.

Reply:
> Actual is better as that's how it behaves for other databases.

Done, also added a line in readme which make it clear to users that all plans are actual query plans

Reply:
Can you please also tell me scope for this PR, would you consider it only when tree visuals are ready?

Reply:
Textual plan is fine. You can do the tree in a separate PR if you want to. I'll review the changes later today.

Reply:
> Textual plan is fine. You can do the tree in a separate PR if you want to. I'll review the changes later today.

Thanks, I shared this with my oracle DBA friends they liked the plan content specially I/O stats, but they complained hitting 5 sec time out more often than not when I looked its mostly in first connection and specially when its remote DB in another region - Like their machine is in ASIA and they are connecting DB in US East.
I see in known issues, need to think how we can get around this.
Just throwing an idea - Few weeks ago I wrote an extension to do Sonar scan in VS with Sonar Qube, To do this I used docked browser control at bottom of VS, Do you think its worth exploring this path?